### PR TITLE
Add xapian-core/1.4.16 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **xapian-core/1.4.16**

Requirement of doxygen

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/xapian-core/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+from conans.errors import ConanInvalidConfiguration
+from contextlib import contextmanager
+import os
+
+
+class XapianCoreConan(ConanFile):
+    name = "xapian-core"
+    description = "Xapian is a highly adaptable toolkit which allows developers to easily add advanced indexing and search facilities to their own applications."
+    topics = ("conan", "xapian", "search", "engine", "indexing", "query")
+    license = "GPL-v2"
```
Comment (uilianries) [inline recipes/xapian-core/all/conanfile.py:11]:
```suggestion
    license = "GPL-2.0-or-later"
```

File: recipes/xapian-core/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        if self.settings.os != "Windows":
+            self.requires("libuuid/1.0.3")
+        self.requires("zlib/1.2.11")
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH") and \
+                tools.os_info.detect_windows_subsystem() != "msys2":
+            self.build_requires("msys2/20190524")
+        # if self.settings.compiler == "Visual Studio":
+        #     self.build_requires("automake/1.16.2")
```
Comment (uilianries) [inline recipes/xapian-core/all/conanfile.py:52]:
```suggestion
```

File: recipes/xapian-core/all/patches/0001-add-msvc-cl-sh.patch [hunk 3]
```diff
@@ -0,0 +1,99 @@
+new file mode 100755
```
Comment (SSE4) [inline recipes/xapian-core/all/patches/0001-add-msvc-cl-sh.patch:1]: I think I have seen that already in another recipe(s). please correct me if I am wrong @madebr 
does it make sense to make it standalone package (build requires)?

Comment (madebr) [reply recipes/xapian-core/all/patches/0001-add-msvc-cl-sh.patch:1]: Indeed, I used it first time in re2c.
It does make sense to create a standalone package for it.
But it's not really well tested. I'm also not sure whether some packages need special cases.

**Review by jgsogo — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Pull request merged**